### PR TITLE
Tests for nested #fun. Issues with LLVM backend

### PR DESCRIPTION
**Pull request by denis-bogdanas**
Tests for nested #fun, incremental increasing of complexity. Work for ocaml, the last test (commented) fails on LLVM backend.

### CODE REVIEW

File: kernel/src/main/java/org/kframework/compile/AddSortInjections.java [hunk 1]
```diff
@@ -186,7 +186,7 @@ private K visitChildren(K term, Sort actualSort) {
                         List<Sort> polySorts = positions.stream()
                                 .map(p -> sort(kapp.items().get(p - 1), freshSortParam))
                                 .collect(Collectors.toList());
-                        expectedSort = lub(polySorts, null, kapp);
+                        expectedSort = lub(polySorts, null, kapp, mod);
```
Comment (traiansf) [inline kernel/src/main/java/org/kframework/compile/AddSortInjections.java:189]: I don't get it: where is the extra argument `mod` being used?

**Review by traiansf — APPROVED** [review]
Looks good to me; I just want some clarification.

File: kernel/src/main/java/org/kframework/compile/AddSortInjections.java (hunk 1 above)
Comment (dwightguth) [reply kernel/src/main/java/org/kframework/compile/AddSortInjections.java:189]: it's being used in upperBounds.

Comment (dwightguth) [reply kernel/src/main/java/org/kframework/compile/AddSortInjections.java:189]: and in the Collection overload of lub.

**Pull request merged**